pkg/actions/service.go

```diff
@@ -326,117 +418,117 @@ func buildRunManifestFromTasks(record graveler.HookRecord, tasks [][]*Task) RunM
+	var v proto.Message = &TaskResultData{}
+	for scanner.Next() {
+		err = scanner.Entry(nil, &v)
 		if err != nil {
-			return nil, fmt.Errorf("get tasks result: %w", err)
+			return err
 		}
-		return nil, nil
-	})
-	if errors.Is(err, db.ErrNotFound) {
-		return ErrNotFound
+		tasks = append(tasks, *taskResultFromProto(v.(*TaskResultData)))
 	}
-	if err != nil || manifest == nil {
+	if err = scanner.Err(); err != nil {
 		return err
 	}
+	manifest.HooksRun = tasks
 
 	// update manifest
 	return s.saveRunManifestObjectStore(ctx, *manifest, storageNamespace, runID)
 }
 
-func (s *Service) GetRunResult(ctx context.Context, repositoryID string, runID string) (*RunResult, error) {
-	res, err := s.DB.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		return s.getRunResultTx(tx, repositoryID, runID)
-	}, db.ReadOnly())
-	if errors.Is(err, db.ErrNotFound) {
-		return nil, fmt.Errorf("run id %s: %w", runID, ErrNotFound)
-	}
+func (s *KVService) deleteRunSecondaryKeys(ctx context.Context, run *RunResultData, repoID string) error {
+	// Delete secondary index by BranchID
+	bk := kv.FormatPath(getRunByBranchPath(repoID, run.BranchId), run.RunId)
+	err := s.Store.Delete(ctx, []byte(bk))
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("delete secondary index by branch (key %s): %w", bk, err)
 	}
-	return res.(*RunResult), nil
-}
 
-func (s *Service) getRunResultTx(tx db.Tx, repositoryID string, runID string) (*RunResult, error) {
-	result := &RunResult{
-		RunID: runID,
-	}
-	err := tx.Get(result, `SELECT event_type, branch_id, source_ref, start_time, end_time, passed, commit_id
-			FROM actions_runs
-			WHERE repository_id=$1 AND run_id=$2`,
-		repositoryID, runID)
+	// Delete secondary index by CommitID
+	ck := kv.FormatPath(getRunByCommitPath(repoID, run.CommitId), run.RunId)
+	err = s.Store.Delete(ctx, []byte(ck))
 	if err != nil {
-		return nil, fmt.Errorf("get run result: %w", err)
+		return fmt.Errorf("delete secondary index by commit (key %s): %w", ck, err)
 	}
-	return result, nil
+
+	return nil
 }
 
-func (s *Service) GetTaskResult(ctx context.Context, repositoryID string, runID string, hookRunID string) (*TaskResult, error) {
-	res, err := s.DB.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		result := &TaskResult{
-			RunID:     runID,
-			HookRunID: hookRunID,
-		}
-		err := tx.Get(result, `SELECT hook_id, action_name, start_time, end_time, passed
-			FROM actions_run_hooks 
-			WHERE repository_id=$1 AND run_id=$2 AND hook_run_id=$3`,
-			repositoryID, runID, hookRunID)
-		if err != nil {
-			return nil, fmt.Errorf("get task result: %w", err)
+func (s *KVService) GetRunResult(ctx context.Context, repositoryID string, runID string) (*RunResult, error) {
+	runKey := getRunPath(repositoryID, runID)
+	m := RunResultData{}
+	_, err := s.Store.GetMsg(ctx, runKey, &m)
+	if err != nil {
+		if errors.Is(err, kv.ErrNotFound) {
+			err = fmt.Errorf("%s, %w", err, ErrNotFound) // Wrap error for compatibility with DBService
```

Comment:
I'm not so sure what's the type of the returned error here. 
```suggestion
			err = fmt.Errorf("%s: %w", err, ErrNotFound) // Wrap error for compatibility with DBService
```